### DISCUSSION TITLE
Installing Homebrew on Linux

### DISCUSSION THREAD

**Discussion by destinyb-realityi** [community/community#25163] [Actions]
<p>I’m attempting to install Homebrew in a Linux build by doing the below: </p>
<pre><code>- name: Installing Homebrew
      run: |
        touch ~/.bash_profile ~/.profile
        /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
        test -d ~/.linuxbrew &amp;&amp; eval $(~/.linuxbrew/bin/brew shellenv)
        test -d /home/linuxbrew/.linuxbrew &amp;&amp; eval $(/home/linuxbrew/.linuxbrew/bin/brew shellenv)
        test -r ~/.bash_profile &amp;&amp; echo "eval \$($(brew --prefix)/bin/brew shellenv)" &gt;&gt; ~/.bash_profile
        echo "eval \$($(brew --prefix)/bin/brew shellenv)" &gt;&gt; ~/.profile
        echo 'export PATH="/home/linuxbrew/.linuxbrew/bin:$PATH"' &gt;&gt; ~/.bash_profile
        source ~/.profile
        brew --version
    - name: Verify Homebrew's Installation
      run: brew --version
</code></pre>
<p>But I get the following error: </p>
<pre><code>line 1: brew: command not found
</code></pre>
<p>I did find out though if I run <em>source ~/.profile &amp;&amp; brew --version</em> in the second step it  <strong>does work</strong>. My question is, is there a way to have <em>source ~/.profile</em> retained between each step in a job? Is there a better way to go about doing this? </p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/destinyb-realityi">@destinyb-realityi</a> ,</p>
<p>It is very strange, I tested the same installation steps on the on <strong>GitHub-hosted Ubuntu runner</strong> and my own  <strong>Ubuntu</strong> machine (a VM), on my own  <strong>Ubuntu</strong>  VM all the steps can work fine without errors, but on  <strong>GitHub-hosted Ubuntu runner</strong>  I also get the same error as your reported.</p>
<p>Below is the steps I use, I did a little changes based on yours:</p>
<pre><code>jobs:
  test:
    name: Test
    runs-on: ubuntu-latest
    steps:
      - name: Installing Homebrew
        run: |
          /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
          test -d ~/.linuxbrew &amp;&amp; eval $(~/.linuxbrew/bin/brew shellenv)
          test -d /home/linuxbrew/.linuxbrew &amp;&amp; eval $(/home/linuxbrew/.linuxbrew/bin/brew shellenv)
          echo "eval \$($(brew --prefix)/bin/brew shellenv)" &gt;&gt;~/.profile
          echo 'export PATH="/home/linuxbrew/.linuxbrew/bin:$PATH"' &gt;&gt;~/.bashrc
          echo 'export PATH="/home/linuxbrew/.linuxbrew/bin:$PATH"' &gt;&gt;~/.profile
          source ~/.bashrc
          source ~/.profile
          brew --version
      - name: Verify Homebrew's Installation
        run: |
          cat ~/.profile
          echo "======================="
          cat ~/.bashrc
          echo "======================="
          brew --version
</code></pre>
<p>You can see in the second step " <strong>Verify Homebrew’s Installation</strong>", I tried to print the contexts of the shell profile scripts, I can confirm the line " <strong>export PATH="/home/linuxbrew/.linuxbrew/bin:$PATH</strong>" indeed has been added into shell profile scripts. But the command " <strong>brew --version</strong>" still always returns " <strong>brew: command not found</strong>".</p>
<p>I have reported this question to the appropriate engineering team for further investigation and evaluation. If they have any progress, I will notify you in time, and sometimes the appropriate engineers may directly reply you here.</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/destinyb-realityi">@destinyb-realityi</a> ,</p>
<p>According to the discussion with the appropriate engineers, the shell profile script ( <strong>.profile</strong> , <strong>.bashrc</strong>  or . <strong>bash_profile</strong> ) is only read for interactive login shells. more details, see <a href="https://www.gnu.org/software/bash/manual/html_node/Bash-Startup-Files.html" rel="nofollow noopener">here</a>.</p>
<p>If you want to add  <strong>Homebrew</strong> to the <strong>PATH</strong> for all the subsequent steps in the same job, you can directly add <strong>Homebrew</strong>’s directory in  <strong>PATH</strong>  via the command " <strong>::add-path::{path}</strong>", reference <a href="https://help.github.com/en/actions/reference/workflow-commands-for-github-actions#adding-a-system-path" rel="nofollow noopener">here</a>:</p>
<pre><code>echo "::add-path::/path/to/dir"
</code></pre>
<p>So the below workflow can work as expected:</p>
<pre><code>jobs:
  test:
    name: Test
    runs-on: ubuntu-latest
    steps:
      - name: Installing Homebrew
        run: |
          /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
          test -d ~/.linuxbrew &amp;&amp; eval $(~/.linuxbrew/bin/brew shellenv)
          test -d /home/linuxbrew/.linuxbrew &amp;&amp; eval $(/home/linuxbrew/.linuxbrew/bin/brew shellenv)
          echo "eval \$($(brew --prefix)/bin/brew shellenv)" &gt;&gt;~/.profile
          echo "::add-path::/home/linuxbrew/.linuxbrew/bin"
          brew --version
      - name: Verify Homebrew's Installation
        run: brew --version
</code></pre>

**Comment by destinyb-realityi** [comment]
<p>Excellent! Thank you very much for your assistance :blush:</p>
